### ISSUE TITLE
Use Symfony events to register registration profiles

### ISSUE THREAD

**Issue by bjendres** [systopia/de.systopia.remoteevent#33]
Currently, all profiles are registered in an option group called ``remote_registration_profiles``. This is not flexible enough, so we want to replace this by a ``RegistrationProfileList`` Symfony event, that allows anyone to register profiles dynamically.

The event should:
- also register the profiles from the old option group, so we're still compatible
- allow the registration of multiple profiles with the same class name, but different init parameters
- (later) add context information (``event_id``, ``event_type_id``, etc.) so you can register certain profiles based on this additional information.

**Comment by dontub** [comment]
This only happens if an intermediate version of this branch was used in which the `_generic` fields were added. These aren't used, anymore, and can be deleted:
```
cv api4 CustomField.delete '{"where":[["name","IN",["remote_registration_default_profile_generic","remote_registration_profiles_generic","remote_registration_default_update_profile_generic","remote_registration_update_profiles_generic"]]]}'
```

**Comment by jensschuppe** [comment]
This has been released with version 1.1-beta1.
